<!--REQUEST_ID=8c31dda0-c01b-4625-9851-9b1671a61645-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.51244&attempt_id=668edfd2-b83c-4482-962e-e43460fa1440)

Attempted: perceptualdiff

<details><summary>Partial log (click to expand)</summary><p>

```
[ 85%] Building CXX object CMakeFiles/perceptualdiff.dir/perceptualdiff.o
[ 85%] Building CXX object CMakeFiles/perceptualdiff.dir/compare_args.o
[100%] Linking CXX executable perceptualdiff
/nix/store/3k95i31hc1cfvv45qgvrpmhvkxlvkg4j-freeimage-3.17.0/lib/libfreeimage.so: undefined reference to `png_init_filter_functions_neon'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/perceptualdiff.dir/build.make:101: perceptualdiff] Error 1
make[1]: *** [CMakeFiles/Makefile2:142: CMakeFiles/perceptualdiff.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
builder for '/nix/store/gimqlwbziy0zgfv4b830vls0fvxfcmmk-perceptualdiff-2.1.drv' failed with exit code 2
error: build of '/nix/store/gimqlwbziy0zgfv4b830vls0fvxfcmmk-perceptualdiff-2.1.drv' failed
```
</p></details>

--- a/pkgs/tools/graphics/perceptualdiff/default.nix
+++ b/pkgs/tools/graphics/perceptualdiff/default.nix
@@ -1,0 +1,25 @@
+{ stdenv, fetchFromGitHub, cmake, freeimage }:
+
+stdenv.mkDerivation rec {
+  pname = "perceptualdiff";
+  name = "${pname}-${version}";
+  version = "2.1";
+
+  src = fetchFromGitHub {
+    owner = "myint";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "176n518xv0pczf1yyz9r5a8zw5r6sh5ym596kmvw30qznp8n4a8j";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ freeimage ];
+
+  meta = with stdenv.lib; {
+    description = "A program that compares two images using a perceptually based image metric";
+    homepage = "https://github.com/myint/perceptualdiff";
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ uri-canva ];
+    platforms = platforms.x86;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -4662,6 +4662,8 @@ with pkgs;
 
   pepper = callPackage ../tools/admin/salt/pepper { };
 
+  perceptualdiff = callPackage ../tools/graphics/perceptualdiff { };
+
   percona-xtrabackup = callPackage ../tools/backup/percona-xtrabackup {
     boost = boost159;
   };

